File: src/main/java/me/aiglez/gangs/commands/JoinCommand.java
package me.aiglez.gangs.commands;

import co.aikar.commands.BaseCommand;
import co.aikar.commands.annotation.*;
import me.aiglez.gangs.gangs.Gang;
import me.aiglez.gangs.users.User;

@CommandAlias("gang")
public class JoinCommand extends BaseCommand {

    @Subcommand("join") @Syntax("<gang>") @CommandCompletion("@gangs_invited")
    public void join(final User user, final Gang gang) {
        if(!gang.isInvited(user)) {
            user.messagec("join.not-invited", gang.getName());
            return;
        }

        if(user.hasGang()) {
            user.messagec("join.already-member");
            return;
        }

        int max = 6;
        if(gang.getMembers().size() == max) {
            user.messagec("join.limit-reached", gang.getName());
            return;
        }

        // booster

        gang.removeInvite(user);
        gang.addMember(user);
        user.setGang(gang);

        user.messagec("join.joined", gang.getName());
    }

    @Subcommand("forcejoin") @Syntax("<gang>") @CommandCompletion("@gangs") @CommandPermission("gang.admin.forcejoin")
    public void forceJoin(final User user, final Gang gang) {
        int max = 6;
        if(gang.getMembers().size() == max) {
            user.messagec("join.limit-reached", gang.getName());
            return;
        }

        gang.addMember(user);
        user.setGang(gang);

        user.message("&c[Admin] Successfully joined {0}", gang.getName());
    }
}

File: src/main/java/me/aiglez/gangs/commands/player/JoinCommand.kt
package me.aiglez.gangs.commands.player

import co.aikar.commands.BaseCommand
import co.aikar.commands.annotation.*
import me.aiglez.gangs.gangs.Gang
import me.aiglez.gangs.helpers.Configuration
import me.aiglez.gangs.helpers.Message
import me.aiglez.gangs.users.User

@CommandAlias("gang")
class JoinCommand : BaseCommand() {

    @Subcommand("join")
    @Syntax("<gang>")
    @CommandCompletion("@gangs_invited")
    fun join(user: User, gang: Gang) {
        if(user.hasGang()) {
            user.message(Message.ALREADY_MEMBER)
            return
        }

        if(gang.isInvited(user)) {
            if(gang.members.size == Configuration.getInteger("max-members")) {
                user.message(Message.JOIN_FULL, gang.name)
                gang.message(Message.JOIN_FULL_ALERT, user.player.name)
                return
            }

            user.gang = gang

            gang.removeInvite(user)
            gang.addMember(user)

            // apply AutoSell multiplier
            gang.core.addBooster(user, gang.core.booster)

            user.message(Message.JOIN_JOINED, gang.name)
            gang.message(Message.JOIN_ANNOUNCEMENT, setOf(user), user.player.name)
        } else {
            user.message(Message.NOT_INVITED, gang.name)
        }
    }

    @Subcommand("forcejoin")
    @Syntax("<gang>")
    @CommandCompletion("@gangs")
    @CommandPermission("gang.admin.forcejoin")
    fun forceJoin(user: User, gang: Gang) {

    }
}